File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.py
class Solution:
    def isPalindrome(self, s: str, i: int, j: int) -> bool:
        while i < j:
            if s[i] == s[j]:
                i+=1
                j-=1
            else:
                return False
        return True
        
    def validPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1
        while i < j:
            if s[i] == s[j]:
                i+=1
                j-=1
            else:
                return self.isPalindrome(s,i,j-1) or self.isPalindrome(s,i+1,j)
        return True

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.cpp
class Solution {
  private:
  bool is_palindrom(string s,int i ,int j){
     
      while(i<j){
          if(s[i]==s[j]){
              i+=1;
              j-=1;
          }
          else{
              return false;
          }
      }
      return true;
  }
    
    
public:
    bool validPalindrome(string s) {
        int i = 0;
        int j = s.size()-1;
        // bool ans;
        while(i<=j){
            if(s[i]==s[j]){
                i+=1;
                j-=1;
            }
            else{
                return (is_palindrom(s,i,j-1)||is_palindrom(s,i+1,j));
            }  
           
        }
        return true;
        
        
    }
};